tests/test_manager_api.py

```diff
@@ -111,6 +111,31 @@ def test_read_file(get_dapp):
             assert dapp.read_file(file_type) == f.read()
 
 
+@pytest.mark.parametrize("get_dapp", get_dapp_scenarios)
+def test_read_file_iter(get_dapp):
+    dapp = get_dapp(
+        [sys.executable, asset_path("write_mock_files.py")], state_file=True, data_file=True
+    )
+    sleep(0.5)
+
+    for file_type in ("state", "data"):
+        mock_fname = asset_path(f"mock_{file_type}_file.txt")
+
+        iterator = dapp.read_file_iter(file_type, False)
+
+        try:
+            with open(mock_fname) as f:
+                assert next(iterator) == f.read()
+
+            with dapp.storage.file_name(file_type).open("a") as file:
+                file.write(f"some\nmore\n{file_type}\nlines!")
+
+            assert next(iterator) == f"some\nmore\n{file_type}\nlines!"
```

Comment:
File content can be moved to a separated variable `f"some\nmore\n{file_type}\nlines!"`